### ISSUE TITLE
How to draw a patient stratification chart.

### ISSUE THREAD

**Issue by sjjadsa** [CYing18/SurvMamba#1]
Dear Author,

Hello! Could you please explain how the figure in your paper was created? For example, what data is required (how to obtain the data and its format) and what tools were used? I would like to create a similar figure using my own dataset. I look forward to your prompt reply. Thank you!

![Image](https://github.com/user-attachments/assets/f4d0b432-57f1-46af-9747-02298ed8df76)

**Comment by CYing18** [comment]
Thank you for your inquiry! To generate a similar figure, I used the Kaplan-Meier survival analysis, which compares the survival distributions of two groups based on predicted scores. The necessary data includes: (1)&nbsp;true_OS_time: the actual survival times. (2)pred_OS_time: the predicted survival times. (3)true_status: the event indicator (1/0).The data format should be a CSV file with these columns. I hope this helps you.










&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: ***@***.***&gt;;
Date: &nbsp;Mon, Feb 17, 2025 11:07 AM
To: ***@***.***&gt;; 
Cc: ***@***.***&gt;; 
Subject: &nbsp;[CYing18/SurvMamba] How to draw a patient stratification chart. (Issue #1)

&nbsp;

   
Dear Author,
 
Hello! Could you please explain how the figure in your paper was created? For example, what data is required (how to obtain the data and its format) and what tools were used? I would like to create a similar figure using my own dataset. I look forward to your prompt reply. Thank you!
 
image.png (view on web)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***&gt;
  sjjadsa created an issue (CYing18/SurvMamba#1)
 
Dear Author,
 
Hello! Could you please explain how the figure in your paper was created? For example, what data is required (how to obtain the data and its format) and what tools were used? I would like to create a similar figure using my own dataset. I look forward to your prompt reply. Thank you!
 
image.png (view on web)

**Comment by sjjadsa** [comment]
Thank you very much for your response! I have two additional questions:
(1) Is the division of the two groups based on the event indicator (1/0)?
(2) In your code, is the output head formatted as follows:
logits = self.classifier(h)  # [B, n_classes]  
Y_prob = F.softmax(logits, dim=1)  
Y_hat = torch.topk(logits, 1, dim=1)[1]  
A_raw = None  
results_dict = None  
if self.survival:  
    Y_hat = torch.topk(logits, 1, dim=1)[1]  
    hazards = torch.sigmoid(logits)  
    S = torch.cumprod(1 - hazards, dim=1)  
    return hazards, S, Y_hat, None, None  
If so, which parameter corresponds to pred_OS_time?
If not, how is pred_OS_time obtained?

Thank you again, and I look forward to your clarification!

**Comment by sjjadsa** [comment]
Dear Author,
Hello! I now understand that you divided all patients into low-risk and high-risk groups based on the median value of the predicted risk scores generated by SurvMamba. However, I am still unclear about which parameter corresponds to the `pred_OS_time` you mentioned earlier. I look forward to your prompt reply. Thank you!
